src/atomate2/abinit/schemas/core.py

```diff
@@ -0,0 +1,125 @@
+"""Core definitions of Abinit calculations documents."""
+
+import logging
+import os
+from pathlib import Path
+from typing import Type, TypeVar, Union
+
+from abipy.abio.inputs import AbinitInput
+from abipy.flowtk import events
+from abipy.flowtk.utils import File
+from jobflow.utils import ValueEnum
+from pydantic import BaseModel, Field
+from pymatgen.core.structure import Structure
+
+from atomate2.abinit.files import load_abinit_input
+from atomate2.abinit.sets.base import AbinitInputSet
+from atomate2.abinit.utils.common import (
+    LOG_FILE_NAME,
+    MPIABORTFILE,
+    OUTPUT_FILE_NAME,
+    get_event_report,
+    get_final_structure,
+)
+from atomate2.common.schemas.structure import StructureMetadata
+
+_T = TypeVar("_T", bound="AbinitTaskDocument")
+
+logger = logging.getLogger(__name__)
+
+
+class Status(ValueEnum):
+    """Abinit calculation state."""
+
+    # TODO: merge this somewhere with vasp => common calculation schema ?
+
+    SUCCESS = "successful"
+    UNCONVERGED = "unconverged"
+    FAILED = "failed"
+
+
+class JobMetadata(BaseModel):
+    """Definition of job metadata fields."""
+
+    dir_name: str = Field(None, description="The directory of this job.")
+    calc_type: str = Field(None, description="The type of calculation of this job.")
+
+
+class AbinitJobSummary(JobMetadata):
+    """Definition of summary information about an Abinit Job."""
+
+    # restart_info: RestartInfo = Field(
+    #     None, description="Restart information for the next job."
+    # )
+    # history: JobHistory = Field(None, description="Job history.")
+    abinit_input_set: AbinitInputSet = Field(
+        None, description="AbinitInputSet object used to perform calculation."
+    )
+    structure: Structure = Field(
+        None, description="Final structure of the calculation."
+    )
+    energy: float = Field(None, description="Final energy of the calculation.")
+
+
+class AbinitTaskDocument(StructureMetadata):
```

Comment:
Yes, related to #145

Reply:
Would it be possible to store some outputs here so these workflows could be used? If you follow the VASP task document roughly that would mean it could be a reasonable template to use for a general task document.

Reply:
Just bumping the above comment.